incorrect definition of esp_app_desc_t.reserv3 fails to compile with ESP IDF v5.4

## Bug description

the definition of esp_app_desc_t seems to be incorrect when used with ESP-IDF 5.4 according to the definition in https://docs.espressif.com/projects/esp-idf/en/v5.4/esp32/api-reference/system/misc_system_api.html#_CPPv4N14esp_app_desc_t7reserv3E

When using the esp_idf_svc::sys::esp_app_desc!() macro I get the following error :
```
error[E0308]: mismatched types
   --> /home/esp/.cargo/registry/src/index.crates.io-6f17d22bba15001f/esp-idf-sys-0.36.1/src/app_desc.rs:102:26
    |
2   | macro_rules! esp_app_desc {
    | -------------------------
    | |
    | in this expansion of `esp_idf_svc::sys::esp_app_desc!` (#1)
    | in this expansion of `$crate::esp_app_desc!` (#2)
...
5   |         $crate::esp_app_desc!(false);
    |         ---------------------------- in this macro invocation (#2)
...
102 |                 reserv3: [0; 4],
    |                          ^^^^^^ expected an array with a fixed size of 3 elements, found one with 4 elements
    |
   ::: src/main.rs:42:5
    |
42  |     esp_idf_svc::sys::esp_app_desc!();
    |     --------------------------------- in this macro invocation (#1)
```

Note: 
https://github.com/esp-rs/esp-idf-sys/blob/e6626272c5ad6038cb10718d716d70326b4027ff/src/app_desc.rs#L105

While the [official definition](https://docs.espressif.com/projects/esp-idf/en/v5.4/esp32/api-reference/system/misc_system_api.html#_CPPv4N14esp_app_desc_t7reserv3E) is
```
uint8_t reserv3[3]
    reserv3
```

- Would you like to work on a fix? [y]

## To Reproduce

use ESP IDF v5.4 and include esp_idf_svc::sys::esp_app_desc!(); in your main()

<!-- Please share the minimal repro of the issue where the bug can be reproduced. -->

<!-- Make sure you are able to reproduce the bug in the `master` branch, too. -->

## Expected behavior

<!-- A clear and concise description of what you expected to happen. Attach screenshots if needed. -->
This could be a fix
https://github.com/mike1703/esp-idf-sys/commit/e06e6ef2fff65dd6bf2cf2346e81e6d7d25670a2

## Environment
- via esp-idf-svc = "0.51.0"
  - => esp-idf-sys  = "0.36.1"
ESP_IDF_VERSION = { value = "tag:v5.4" }

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Fixed
+- Fix the `esp_app_desc!` macro so the reserv3 field is correct for ESP IDF v5.4
 - Fix the `esp_app_desc!` macro so that it properly embeds `CONFIG_ESP_EFUSE_BLOCK_REV_MIN_FULL`/`CONFIG_ESP_EFUSE_BLOCK_REV_MAX_FULL` with recent ESP-IDF versions (5.4+, 5.2.3+, 5.3.2+)
 - Make sure that the `esp_app_desc!` macro fails at build-time if some of the fields cannot fit inside the structure
 - Re-enable "esp_lcd_panel_rgb.h" after >= v5.3 for targets with SOC_LCD_RGB_SUPPORTED

--- a/src/app_desc.rs
+++ b/src/app_desc.rs
@@ -102,7 +102,7 @@ macro_rules! esp_app_desc {
                     esp_idf_version = "5.2",
                     esp_idf_version = "5.3"
                 )))]
-                reserv3: [0; 4],
+                reserv3: [0; 3],
                 #[cfg(not(any(
                     esp_idf_version_major = "4",
                     esp_idf_version = "5.0",

